docs/contributing/testing.md

```diff
@@ -0,0 +1,114 @@
+
+## 4. Run the automated tests
+
+In your project folder, run `npm test` to run the automated tests, including linting.
+
+If a test fails, you should check your code for any errors, then fix any tests you need to.
+
+## 5. Write new tests
+
+You should write new tests if you’ve created a new component, or changed the way a component works by:
+
+- changing or adding to its JavaScript code
+- changing or adding to its Nunjucks macro
+- creating or updating a Sass mixin or function
+
+If you’re not sure how to write new tests, ask a developer on your team to help you or [contact us](https://design-system.service.gov.uk/#support).
+
+### If you created a component
+
+Create the following files in the `src/govuk/components` folder:
+
+- `<COMPONENT>/<COMPONENT>.test.js` - to test functionality if the component uses JavaScript
+- `<COMPONENT>/template.test.js` - to test the Nunjucks macro
+
+Where `<COMPONENT>` is the name of the component you created.
+
+You can use the existing files in the `src/govuk/components` folder as templates for your new files.
+
+### If you changed or added to a component
+
+In the `src/govuk/components` folder, update or add tests to:
+
+- `<COMPONENT>/<COMPONENT>.test.js` - if you updated functionality
+- `<COMPONENT>/template.test.js` - if you updated the Nunjucks macro
+
+Where `<COMPONENT>` is the name of the component you changed or added to.
+
+### If you created or updated a Sass mixin or function
+
+Update or add tests in `src/govuk/<LAYER>/<LAYER>.test.js`, where `<LAYER>` is the [layer](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/README.md#structure) you’re creating or updating a mixin for].
+
+## 6. Update the snapshot tests
+
+If your component uses another component, one of the [Jest snapshot tests](https://jestjs.io/docs/en/snapshot-testing) may fail. Snapshot tests compare a component's current markup with a previously stored version.
+
+If a snapshot test fails, do the following.
+
+1. Check that the component's new markup is correct.
+2. Run `npm test -- -u src/govuk/components/<COMPONENT>` to update the snapshot test with the new markup.
+3. Commit the updated file in the `/src/govuk/components/<COMPONENT>/__snapshots__/` folder.
+4. In the commit message, tell us you're updating the snapshot file and why.
+
+Where `<COMPONENT>` is the name of the component you changed.
+
+## 7. Tell us what you’ve checked
+
+When you create the pull request for your contributions, list what you’ve tested and checked in the pull request description.
+
+If your contribution changes how a component looks, include before and after screenshots if you can.
+
+### If GitHub shows the build failed
+
+1. At the bottom of your pull request in GitHub, find the line that says **Build Failed**.
+2. Select **Details**.
+3. Under **Build Failed**, select **The build**.
+4. Wait for the build log to appear.
+5. Use the information in the build log to find and fix your code.
```

Comment:
Does the information in the build log really help you find… your code?

Reply:
Hmm good point - seems like we've lost a couple of words somewhere! Will fix - thanks!